hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/validate/operators/typeinference/HazelcastOperandTypeInference.java

```diff
@@ -61,11 +64,13 @@ public void inferOperandTypes(SqlCallBinding callBinding, RelDataType returnType
                 JetTableFunctionParameter parameter = parametersByName.get(name);
                 if (parameter != null) {
                     SqlTypeName parameterType = parameter.type();
-                    operandTypes[i] = toType(parameterType, typeFactory);
+                    parameterTypes[parameter.ordinal()] = toType(parameterType, typeFactory);
                 } else {
                     throw SqlUtil.newContextException(id.getParserPosition(), RESOURCE.unknownArgumentName(name));
                 }
             }
+            //noinspection ResultOfMethodCallIgnored
+            Arrays.stream(parameterTypes).filter(Objects::nonNull).toArray(ignored -> operandTypes);
```

Comment:
What does this expression do actually?

Reply:
Copies over non-null elements. `callBinding` & `operandTypes` respectively contains and has capacity for actually used parameters.

Reply:
I'd also prefer to see this as a for loop, this is hard to read...